### PR TITLE
Enable testing index Weaviate

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
PR that enables testing the weaviate indexing component 

Leverage the embedded weaviate component to setup the client and bypass the docker-compose setup 

https://weaviate.io/developers/weaviate/installation/embedded

### CODE REVIEW

File: components/index_weaviate/tests/test_component.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    )
+
+    result = query.do()
+
+    return result["data"]["Get"][class_name]
+
+
+def test_index_weaviate_component(monkeypatch):
+    client = weaviate.Client(embedded_options=EmbeddedOptions())
+    monkeypatch.setattr("weaviate.Client", lambda *args, **kwargs: client)
```
Comment (RobbeSneyders) [inline components/index_weaviate/tests/test_component.py:33]: Why is this needed? I didn't need this in the [retrieval component test](https://github.com/ml6team/fondant/blob/main/components/retrieve_from_weaviate/tests/test_component.py).

Comment (PhilippeMoussalli) [reply components/index_weaviate/tests/test_component.py]: Missed the fact that we have access to a local URL, adjusted!

**Review by RobbeSneyders — APPROVED** [review]
Thanks!
